### PR TITLE
feat: refactor modals into general component

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Create generic status modal](https://www.notion.so/uwblueprintexecs/Create-generic-status-modal-9f7808a0a41c4919ab0acd579e4ae069?pvs=4)

## Implementation description
* Refactored AddAdminModal and ResponseTypeModal to use a generic component
* Added ArchiveModal as an example Status Modal

## Steps to test
image
image
image
image

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ const App = (): React.ReactElement => {
                   component={TeacherPage}
                   exact
                   path={Routes.TEACHER_LANDING}
-                  roles={["Teacher"]}
+                  roles={["Teacher", "Admin"]}
```
Comment (carissa-tang) [inline frontend/src/App.tsx:85]: does it make sense to allow admins to see the teacher page?

Comment (joyce-shi) [reply frontend/src/App.tsx:85]: no, i think we just left that in there for now so you dont have to logout to see other pages

**Review by joyce-shi — APPROVED** [review]
LGTM!

File: frontend/src/App.tsx (hunk 1 above)
Comment (carissa-tang) [reply frontend/src/App.tsx:85]: wait like currently its only for teachers - but i'll add back admins for now lol

Comment (carissa-tang) [reply frontend/src/App.tsx:85]: but also shouldn't admins have access the everything?

**Pull request merged**